Code/GraphMol/CIPLabeler/CIPLabeler.cpp

```diff
@@ -0,0 +1,203 @@
+//
+//
+//  Copyright (C) 2020 Schrödinger, LLC
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+#include <algorithm>
+
+#include <GraphMol/RDKitBase.h>
+
+#include "CIPLabeler.h"
+#include "CIPMol.h"
+#include "configs/Sp2Bond.h"
+#include "configs/Tetrahedral.h"
+
+#include "rules/Rules.hpp"
+#include "rules/Rule1a.h"
+#include "rules/Rule1b.h"
+#include "rules/Rule2.h"
+#include "rules/Rule3.h"
+#include "rules/Rule4a.h"
+#include "rules/Rule4b.h"
+#include "rules/Rule4c.h"
+#include "rules/Rule5New.h"
+#include "rules/Rule6.h"
+
+namespace RDKit {
+namespace CIPLabeler {
+
+namespace {
+std::vector<Configuration *> findConfigs(CIPMol &mol) {
+  std::vector<Configuration *> configs;
+
+  for (auto &atom : mol.atoms()) {
+    auto chiraltag = static_cast<int>(atom->getChiralTag());
+    if (chiraltag > 0) {
+      auto carriers = std::vector<Atom *>();
+      carriers.reserve(4);
+      for (auto &bond : mol.getBonds(atom)) {
+        auto nbr = bond->getOtherAtom(atom);
+        carriers.push_back(nbr);
+      }
+      if (carriers.size() < 4) {
+        // Implicit H -- use the central atom instead of a dummy H
```

Comment:
I take it is a "cheap" way of always having a fourth atom to use in `Configuration::parity4()´ in Tetrahedral configurations.

Since it is only used in equality comparisons, I think it might also be possible to use a `nullptr`, but in this way we get some "protection" for the case that something goes wrong, and we accidentally get a non-intentional `nullptr` passed into the vector that goes into `Configuration::parity4()´.

Reply:
It's hard to tell from this code, but are you assuming that after this block of code `carriers` will always have 4 elements?

This isn't always true. For example, three coordinate chiral centers with an implicit H may only have 2 actual neighbors. The RDKit, following InChI, specifically recognizes phosphines - `Cl[P@H]F` - and arsines `Cl[As@H]F`

Reply:
Good catch. For arsines, phospines, and other trigonal pyramid configurations we need to add a 4th reference, representing a "phantom atom" that completes the pseudo-tetrahedral configuration, so that it can be resolved like "true" tetrahedrals.